geoh5py/data/data.py

```diff
@@ -62,12 +62,11 @@ def __init__(
         if self.entity_type.name == "Entity":
             self.entity_type.name = self.name
 
-        data_type.workspace._register_data(self)
+        getattr(data_type.workspace, "_register_data")(self)
```

Comment:
We've been warned by @sebhmg to avoid sweeping the private methods under the rug with `getattr`.

@sebhmg Would there be an issue with making the registering public? Else, we could add a `register_entity` method that calls the right private method depending on class type...

Reply:
Ok @MatthieuCMira , let's create a public `register` method on the workspace. Then workspace does the `weakref_utils.insert_once` based on the type of the incoming argument. We can simplify the inits of all other class types later, but do the change here now.

Reply:
right, since each object registers upon its instantiation, it probably make sense to expose it as a public method.
Discussing with @domfournier, he suggest a single public `register()` method that would call the other private ones base on the object type. It seems to make sense

Reply:
Should look here like

```suggestion
        data_type.workspace.register(self)
```

Reply:
does it need to be in each Entity class then?
Could it be in Entity __init__ ?

Reply:
Exactly what I had in mind. We can move the registering all the way down the base Entity class + property_groups (which are special kind). Same for entity_type

Reply:
I created a register function and integrated it in entity,
Removed it everywhere else

Reply:
Resolving this conversation